OpenSim/Tools/InverseKinematicsTool.cpp

```diff
@@ -362,17 +362,17 @@ bool InverseKinematicsTool::run()
 
             }
 
-            kinematicsReporter.step(s, i);
+            kinematicsReporter->step(s, i);
             analysisSet.step(s, i);
         }
 
         // Do the maneuver to change then restore working directory 
         // so that output files are saved to same folder as setup file.
         if (_outputMotionFileName!= "" && _outputMotionFileName!="Unassigned"){
-            kinematicsReporter.getPositionStorage()->print(_outputMotionFileName);
+            kinematicsReporter->getPositionStorage()->print(_outputMotionFileName);
         }
         // Once done, remove the analysis we added
-        _model->removeAnalysis(&kinematicsReporter);
+        _model->removeAnalysis(kinematicsReporter);
```

Comment:
Are you sure this memory isn't being leaked?

Reply:
Yes because the problem was that the delete operator was called on kinematicsReporter

https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/ArrayPtrs.h#L752